connect/api.go

```diff
@@ -10,6 +11,20 @@ const (
 	APIVersion = "v4"
 )
 
+func announceSystem(body []byte) (string, string, error) {
```

Comment:
Most of our API calls use some struct as parameter. Why not pass `hwinfo` and string hostname here?

Reply:
announce_system and update_system both build the exact same payload:
https://github.com/SUSE/connect/blob/master/lib/suse/connect/api.rb#L55-L82

This is even longer in Go. I have extracted that to one place
https://github.com/SUSE/connect-ng/pull/9/commits/17237d5f655ecbab8dc5df3ebf4c50b23e221a3d#diff-1fefdd8a034af76909710e05c9949c26981b14a4f065c9cec8b9b6811bc14d9cR72

Reply:
this could still be called inside `announceSystem()` and `updateSystem()` accordingly.

Reply:
why?

Reply:
if you call `makeSysInfoBody()` inside these functions then you can pass relevant data to them instead of raw bytes.

Reply:
sure. But makeSysInfoBody() makes calls to the system, and I want to unit test announceSystem()

Reply:
ah. the old mocking problem again. can you do it like with `var execCommand = exec.Command` ?

Reply:
there are many calls. Why is receiving a slice header a problem here?

Reply:
because looking at the function you have no idea what kind of input it expects.

Reply:
I added a comment. The caller AnnounceSystem() in client.go will be calling makeSysInfoBody()

Reply:
OK.

Reply:
one more thing... maybe mention `makeSysInfoBody()` in the comment as well.